module/zfs/dsl_scan.c

```diff
@@ -141,9 +142,9 @@ dsl_scan_init(dsl_pool_t *dp, uint64_t txg)
 		 */
 		if (err == EOVERFLOW) {
```

Comment:
I wonder if adding a new value to `dsl_scan_phys_t` (and bumping `SCAN_PHYS_NUMINTS` to 25) will prevent us from detecting errata 2094 in the first place, since `zap_lookup` will not EOVERFLOW if the pool is affected.